pkg/admission/admission_controller.go

```diff
@@ -194,7 +194,16 @@ func (c *AdmissionController) processPod(req *admissionv1.AdmissionRequest, name
 		log.Log(log.Admission).Info("bypassing namespace", zap.String("namespace", namespace))
 		return admissionResponseBuilder(uid, true, "", nil)
 	}
-	patch = updateSchedulerName(patch)
+
+	// Only override default-scheduler unless override is enabled
+	if c.conf.GetOverrideCustomSchedulers() && pod.Spec.SchedulerName != constants.SchedulerName {
```

Comment:
See earlier comments: this is not correct, default behaviour should not change. can be turned off if wanted. 2nd part of the check is not needed as we preserve existing behaviour